src/test/xmeta/test_lexer.cpp

```diff
@@ -191,7 +197,9 @@ TEST_CASE("Enum illegal assignments")
 
     xmeta_idl_reader reader{ "" };
     xlang_test_listener listener;
-    REQUIRE(reader.read(test_idl_string_assignment, listener, true) == 1);
+    reader.read(test_idl_string_assignment, listener, true);
+    REQUIRE(reader.get_num_syntax_errors() == 1);
```

Comment:
Consider validating that the error is the one that you expect.

Reply:
Fixing in later PR.